README.md

```diff
@@ -138,43 +167,93 @@ This endpoint returns a list of all existing categories. It will return an empty
 This endpoint returns one category corresponding to the id provided in the request. If no category with that
 id exists, it will return a 404 response code.
 
+#### POST /categories
+
+This endpoint creates a new category. It requires [authentication](#Authentication). It accepts the following
+parameters in the request body as JSON:
+
+|  Name      |   Type  |    Notes    |
+|------------|---------|-------------|
+| title      | String  | *Optional*. |
+| help_text  | String  | *Optional*. |
+| active     | Boolean | *Optional*. Defaults to `true`. Passing in `false` will create a deactivated category. A category cannot be reactivated once it has been deactivated. |
+
+#### PUT /categories/{id}
+
+This endpoint updates an existing category. It requires [authentication](#Authentication). It accepts the following
+parameters in the request body as JSON:
+
+|  Name      |   Type  |    Notes    |
+|------------|---------|-------------|
+| title      | String  | *Optional*. |
+| help_text  | String  | *Optional*. |
+| active     | Boolean | *Optional*. Passing in `false` will deactivate the category. A category cannot be reactivated once it has been deactivated. |
+| subcategories | Array<Object> | *Optional*. See below for information on formatting this parameter. |
+
+The `subcategories` parameter allows you to create subcategories and criteria along with a category. The `subcategories` parameter must be formatted as an array of objects in the following format:
+
+```
+[
+  {
+    id: ...,
+    title: ...,
+    help_text: ...,
+    active: ...,
+    criteria: [
+      {
+        id: ...,
+        title: ...,
+        recommendation_text: ...,
+        help_text: ...,
+        active: ...
+      },
+      ...
+    ]
+  },
+  ...
+]
```

Comment:
@brittanylau Also related to what we were talking about on Slack. Here's the workflow I propose:

* When a user creates a new category, the CMS calls `POST /categories` and returns an empty category with an id.
* Every time the user creates a new subcategory, the CMS calls `POST /subcategories`, and same with criteria.
* When the user hits "save", all of these resources are updated with one PUT call.

This is slightly more complicated than the other endpoint because categories, subcategories, and criteria don't overwrite themselves. They're meant to be referred to by unique ids and there should only be one copy of them. I worry that handling all the create vs. update logic in one big complicated endpoint might result in us creating some objects that already exist or trying to update objects that don't exist. If we separately handle creating and updating, then we can make our updating logic relatively simple.

Reply:
Totally agree! This is a really neat solution :)

Just to make sure I've understood, this mean creating categories/subcategories/criteria will have their own "save" button to trigger the POST call?

Reply:
We could do that, or we could do an auto-save as soon as someone creates a new empty category/subcategory/criterion... but that would be inconsistent if we're not doing auto-save for the whole thing... do you think it would be better to implement multiple little save buttons or just try to do auto-save off the bat?

Reply:
Any thoughts on which one's easier for the frontend? I'm not sure myself, but both approaches sound logical to me!

After thinking it over, I think if we did want to combine everything to be one big PUT, we could use our `update_or_create` functions to nicely handle finding a valid `id`... but I can see how that could get confusing.